org.eclipse.jdt.ls.product/scripts/jdtls

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+import glob
+import os
+import platform
+import re
+import semver
+import subprocess
+import sys
+import xml.etree.ElementTree as ET
+from pathlib import Path
+
+def get_java_executable():
+	java_executable = 'java'
+
+	if 'JAVA_HOME' in os.environ:
+		java_exec_to_test = os.path.join(os.environ['JAVA_HOME'], 'bin', 'java')
+		if os.path.exists(java_exec_to_test):
+			java_executable = java_exec_to_test
+
+	p = subprocess.Popen(java_executable + ' -version', shell = True, stdout = subprocess.PIPE, stderr = subprocess.PIPE, universal_newlines=True)
+	p.wait()
+	out, err = p.communicate()
+
+	matches = re.finditer(r"\d+\.\d+\.\d+", out + err)
+	for matchNum, match in enumerate(matches):
+		java_version = semver.VersionInfo.parse(match.group())
+
+		if java_version.major < 11:
+			raise Exception("jdtls requires at least Java 11")
+
+		return java_executable
+
+	raise Exception("Could not determine Java version")
+
+def find_equinox_launcher(script_path):
+	plugins_dir = os.path.join(os.path.dirname(script_path), "plugins")
+	launchers = glob.glob('org.eclipse.equinox.launcher_*.jar', root_dir = plugins_dir)
+	if len(launchers) > 0:
+		return os.path.join(plugins_dir, launchers[0])
+
+	raise Exception("Cannot find equinox launcher")
+
+def get_shared_config_path(script_path):
+	system = platform.system()
+
+	if system == 'Linux':
+		config_dir = 'config_linux'
+	elif system == 'Darwin':
+		config_dir = 'config_mac'
+	elif system == 'Windows':
+		config_dir = 'config_win'
+
+	return os.path.join(os.path.dirname(script_path), config_dir)
+
+def find_maven_repository_path():
+	settings_xml = os.path.join(Path.home(), '.m2', 'settings.xml')
+
+	if os.path.exists(settings_xml):
+		settings = ET.parse(settings_xml)
```

Comment:
Is it safe to use ElementTree here to parse this file with the potential for injecting malicious XML? From the docs:

> The [xml.etree.ElementTree](https://docs.python.org/3/library/xml.etree.elementtree.html#module-xml.etree.ElementTree) module is not secure against maliciously constructed data. If you need to parse untrusted or unauthenticated data see [XML vulnerabilities](https://docs.python.org/3/library/xml.html#xml-vulnerabilities).

https://docs.python.org/3/library/xml.etree.elementtree.html

Reply:
Switched to `defusedxml`.